packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerModule.java

```diff
@@ -0,0 +1,143 @@
+import java.util.HashMap;
+import java.util.Map;
+
+import expo.core.ExportedModule;
+import expo.core.ModuleRegistry;
+import expo.core.Promise;
+import expo.core.interfaces.ExpoMethod;
+import expo.core.interfaces.ModuleRegistryConsumer;
+import expo.interfaces.taskManager.TaskServiceInterface;
+import expo.interfaces.taskManager.TaskManagerInterface;
+
+public class TaskManagerModule extends ExportedModule implements ModuleRegistryConsumer {
+  static String EVENT_NAME = "TaskManager.executeTask";
+  static String E_TASK_SERVICE_NOT_FOUND = "E_TASK_SERVICE_NOT_FOUND";
+
+  private TaskServiceInterface mTaskService;
+  private TaskManagerInterface mTaskManagerInternal;
+
+  public TaskManagerModule(Context context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return "ExpoTaskManager";
+  }
+
+  @Override
+  public Map<String, Object> getConstants() {
+    Map<String, Object> constants = new HashMap<>();
+    constants.put("EVENT_NAME", EVENT_NAME);
+    return constants;
+  }
+
+  @Override
+  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+    mTaskService = moduleRegistry.getSingletonModule("TaskService", TaskServiceInterface.class);
+    mTaskManagerInternal = moduleRegistry.getModule(TaskManagerInterface.class);
+  }
+
+  //region Expo methods
+
+  @ExpoMethod
+  public void notifyTaskDidFinish(String taskName, Map<String, Object> response, final Promise promise) {
+    if (!checkTaskService(promise)) {
+      return;
+    }
+    mTaskService.notifyTaskDidFinish(taskName, getAppId(), response);
+    promise.resolve(null);
+  }
+
+  @ExpoMethod
+  public void isTaskRegisteredAsync(String taskName, final Promise promise) {
+    if (!checkTaskService(promise)) {
+      return;
+    }
+    promise.resolve(mTaskService.hasRegisteredTask(taskName, getAppId()));
+  }
+
+  @ExpoMethod
+  public void getTaskOptionsAsync(String taskName, final Promise promise) {
+    if (!checkTaskService(promise)) {
+      return;
+    }
+    promise.resolve(mTaskService.getTaskOptions(taskName, getAppId()));
+  }
+
+  @ExpoMethod
+  public void getRegisteredTasksAsync(final Promise promise) {
+    if (!checkTaskService(promise)) {
+      return;
+    }
+    promise.resolve(mTaskService.getTasksForAppId(getAppId()));
+  }
+
+  @ExpoMethod
+  public void unregisterTaskAsync(String taskName, final Promise promise) {
+    if (!checkTaskService(promise)) {
+      return;
+    }
+    try {
+      mTaskService.unregisterTask(taskName, getAppId(), null);
+      promise.resolve(null);
+    } catch (Exception e) {
+      promise.reject(e);
+    }
+  }
+
+  @ExpoMethod
+  public void unregisterAllTasksAsync(final Promise promise) {
+    if (!checkTaskService(promise)) {
+      return;
+    }
+    try {
+      mTaskService.unregisterAllTasksForAppId(getAppId());
+      promise.resolve(null);
+    } catch (Exception e) {
+      promise.reject(e);
+    }
+  }
+
+  @ExpoMethod
+  public void startObserving(final Promise promise) {
```

Comment:
The observers are used by `expo-react-native-adapter`, which assumes these methods exist in order to tell a native module to provide events. It's currently inconsistent across platforms and is an Android-only method.

Reply:
@ide almost everything is correct 😄 our `EventEmitter` in `expo-react-native-adapter` calls these methods only on Android because on iOS they are being called by React Native, so they are consistent across platforms 😉 See https://github.com/facebook/react-native/blob/1151c096dab17e5d9a6ac05b61aacecd4305f3db/React/Modules/RCTEventEmitter.m#L82